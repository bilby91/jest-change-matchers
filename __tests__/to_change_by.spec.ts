import "jest"
import "../"

let sharedState = 0

describe("toChangeBy", () => {
  beforeEach(() => {
    sharedState = 0
  })

  describe(".toChangeBy", () => {
    it("passes with positive changes", async () => {
      await expect(async () => {
        sharedState += 1
      }).toChangeBy(async () => {
        return sharedState
      }, 1)
    })

    it("passes with negative changes", async () => {
      await expect(async () => {
        sharedState -= 1
      }).toChangeBy(async () => {
        return sharedState
      }, -1)
    })

    it("throws an error when it doesn't change the desired amount", async () => {
      const testCase = () => {
        return expect(async () => {
          sharedState += 2
        }).toChangeBy(async () => {
          return sharedState
        }, 1)
      }

      await expect(testCase()).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe(".not.toChangeBy", () => {
    it("passes with positive changes", async () => {
      await expect(async () => {
        sharedState += 1
      }).not.toChangeBy(async () => {
        return sharedState
      }, -1)
    })

    it("passes with negative changes", async () => {
      await expect(async () => {
        sharedState -= 1
      }).not.toChangeBy(async () => {
        return sharedState
      }, 1)
    })

    it("throws an error when it changes the desired amount", async () => {
      const testCase = () => {
        return expect(async () => {
          sharedState += 2
        }).not.toChangeBy(async () => {
          return sharedState
        }, 2)
      }

      await expect(testCase()).rejects.toThrowErrorMatchingSnapshot()
    })
  })
})
