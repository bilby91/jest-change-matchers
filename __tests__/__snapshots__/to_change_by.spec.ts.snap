// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toChangeBy .not.toChangeBy throws an error when it changes the desired amount 1`] = `
"Expected target not to have changed by:
  2
Received:
  2
"
`;

exports[`toChangeBy .toChangeBy throws an error when it doesn't change the desired amount 1`] = `
"Expected target to have changed by:
  1
Received:
  2
"
`;
